web-handlers.go

```diff
@@ -393,7 +393,7 @@ func (web *webAPIHandlers) Upload(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if eventN.IsBucketNotificationSet(bucket) {
+	if eventN != nil && eventN.IsBucketNotificationSet(bucket) {
```

Comment:
There are many places this has to be done. Better to move this one into the 'IsBucketNotificationSet' itself where you can verify. 

https://play.golang.org/p/T_WgFhGxwY